Tools/Scripts/webkitpy/api_tests/runner.py

```diff
@@ -210,6 +214,17 @@ def _run_single_test(self, binary_name, test):
                 stdout_line, stderr_line = server_process.read_either_stdout_or_stderr_line(started + self._timeout)
                 if not stderr_line and not stdout_line:
                     break
+                if stdout_line:
+                    line_count += 1
+                if stderr_line:
+                    line_count += 1
+                if line_count > self.log_limit:
+                    stderr_line = self.EXCEEDED_LOG_LINE_MESSAGE.format(self.log_limit)
+                    stderr_buffer += stderr_line
+                    _log.error(stderr_line[:-1])
+                    server_process.stop()
```

Comment:
Out of curiosity, is this a SIGKILL or something more polite?

Reply:
It's `SIGKILL`, but I think that's appropriate in this case.